Library/Homebrew/cmd/test-bot.rb

```diff
@@ -112,33 +112,37 @@ def run
         return
       end
 
+      @output = ""
       start_time = Time.now
-
       log = log_file_path
-
-      pid = fork do
-        File.open(log, "wb") do |f|
-          STDOUT.reopen(f)
-          STDERR.reopen(f)
-        end
-        Dir.chdir(@repository) if @command.first == "git"
-        exec(*@command)
+      verbose = ARGV.verbose?
+      success = nil
+      working_dir = @command.first == "git" ? @repository : Dir.pwd
+      logf = File.open(log, "wb")
+      puts if verbose
+
+      IO.popen(@command, :err => [:child, :out], :chdir => working_dir) do |pipe|
+         while buf = pipe.gets
+           puts buf if verbose
+           logf.puts buf
+           @output += buf
+         end
+         Process.wait(pipe.pid)
+         success = $?.success?
+         pipe.close
       end
-      Process.wait(pid)
 
       @time = Time.now - start_time
-
-      @status = $?.success? ? :passed : :failed
+      @status = success ? :passed : :failed
       puts_result
 
-      if File.exist?(log)
-        @output = fix_encoding File.read(log)
-        if has_output? and (failed? or @puts_output_on_success)
-          puts @output
-        end
-        FileUtils.rm(log) unless ARGV.include? "--keep-logs"
+      if has_output? && (failed? or @puts_output_on_success) && !verbose
+        puts @output
       end
 
+      logf.flush
+      logf.close
+      FileUtils.rm(log) unless ARGV.include? "--keep-logs"
```

Comment:
Seems weird to write a file and then delete it (although I know that's my existing code). Maybe just read the output into a string buffer and only write it to disk if we want to keep it?